implementations/rust/ockam/ockam_command/src/vault/create.rs

```diff
@@ -38,8 +44,35 @@ async fn run_impl(
     opts: CommandGlobalOpts,
     cmd: CreateCommand,
 ) -> crate::Result<()> {
+    let name = cmd
+        .name
+        .clone()
+        .unwrap_or_else(|| hex::encode(random::<[u8; 4]>()));
+    if cmd.has_default_args() {
+        if let ConfirmResult::No = opts.terminal.confirm(&fmt_info!(
+            "This will create a vault with name '{name}'. Do you want to continue?"
+        ))? {
+            return Ok(());
+        }
+    }
```

Comment:
I'm not 100% convinced about relying on whether the command is being used with all its defaults or not to show the confirmation prompt. Especially because it forces us to remove the clap's attributes. 

How do you think we should handle this? Show it always no matter the arguments being used?

Reply:
I just saw your comment above @mrinalwadhwa (https://github.com/build-trust/ockam/pull/4652#issuecomment-1504510714). I'll leave this open to bring the reviewers' attention to it in the mean time.